packages/lab/src/list-next/Highlighter.tsx

```diff
@@ -0,0 +1,50 @@
+import { makePrefixer } from "@salt-ds/core";
+import { ReactElement } from "react";
+import { escapeRegExp } from "../utils";
+import { useWindow } from "@salt-ds/window";
+import { useComponentCssInjection } from "@salt-ds/styles";
+import highligherCss from "./Highlighter.css";
+
+const withBaseName = makePrefixer("saltHighlighter");
+
+export interface HighlighterProps {
+  matchPattern?: RegExp | string;
+  text?: string;
+}
+
+export const Highlighter = ({
```

Comment:
re the PR description; will this be used in other components?

Reply:
I don't think there are any others that are using it outside list, but we could move it to a more general place if we get a case for it to be reused :)